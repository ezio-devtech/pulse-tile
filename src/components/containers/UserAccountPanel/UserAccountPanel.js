import React from 'react';
import Dropdown from 'simple-react-dropdown'

import PTButton from '../PTButton/PTButton';
import UserPanelItem from '../UserPanelItem/UserPanelItem';
import NotificationContent from '../../representation/temprorary/NotificationContent'

class UserAccountPanel extends React.Component {
  render () {
    return(
      <ul className="user-panel" role="tablist">
        <UserPanelItem className="user-panel-item visible-xs"
           children={(
             <PTButton className="btn-header"
               children={(
                 <i className="fa fa-search"/>
               )}
             />
           )}
        />
        <UserPanelItem className="user-panel-item dropdown"
           children={(
             <Dropdown content={<NotificationContent />}>
               <PTButton className="btn-header btn-notification"
                 children={(
                   <div>
                     <i className="fa fa-bell-o"/>
                     <span className="count">2</span>
                   </div>
                 )}
               />
             </Dropdown>
           )}
        />
        <UserPanelItem className="user-panel-item"
           children={(
             <PTButton className="btn-header btn-user"
               children={(
                 <i className="fa fa-user"/>
               )}
             />
           )}
        />
      </ul>
    )
  }
}

export default UserAccountPanel;