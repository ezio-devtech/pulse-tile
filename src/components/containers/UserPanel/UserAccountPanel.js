import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { lifecycle } from 'recompose';
import { get } from 'lodash';

import packageJSON from '../../../../package.json';
import { testConstants, isDevMode } from '../../../config/for-test.constants';
import PTButton from '../../ui-elements/PTButton/PTButton';
import UserAccountPanelSettings from './UserAccountPanelSettings';
import userAccountSelector from './selectors';
import { initialiseSelector } from '../App/selectors';
import { clientUrls } from '../../../config/client-urls.constants';
import { unmountOnBlur } from '../../../utils/HOCs/unmount-on-blur.utils';
import { logoutStart } from '../../../ducks/logout.duck';
import { themeConfigs } from '../../../themes.config';
import { getDDMMMYYYY } from '../../../utils/time-helpers.utils';

const mapDispatchToProps = dispatch => ({ actions: bindActionCreators({ logoutStart }, dispatch) });

@connect(initialiseSelector)
@connect(userAccountSelector, mapDispatchToProps)
@lifecycle(unmountOnBlur)
export default class UserAccountPanel extends PureComponent {

  static propTypes = {
    user: PropTypes.shape().isRequired,
    actions: PropTypes.objectOf(PropTypes.func).isRequired,
  };

  static contextTypes = {
    router: PropTypes.shape({
      history: PropTypes.object,
    }),
  };

  renderToProfile() {
    this.context.router.history.push(clientUrls.USER_PROFILE);
  }

  render() {
    const { user, actions, onClick, onClose, initialiseData } = this.props;
    const varsionOfPulseTile = packageJSON.version;
    const varsionOfReact = packageJSON.dependencies.react;
    const imageLocation = '/images/user.jpg';
    const imageSource = isDevMode ? (testConstants.hostName + imageLocation) : imageLocation;
    return (
      <div className="dropdown-user dropdown-menu-right dropdown-menu" id="userAccountPanelElement" >
        { themeConfigs.isShowUserPhoto ?
            <div className="user-profile-image" onClick={() => this.renderToProfile()}>
              <div className="img">
                <img src={imageSource} alt="" />
              </div>
            </div>
        : null }
        <div className="user-profile-info">
          <div
            className="user-profile-info__item name"
            onClick={() =>{
              this.context.router.history.push(clientUrls.USER_PROFILE);
              onClick('')
            }}
          >{user.given_name} {user.family_name}</div>
          <div className="user-profile-info__descr">
            <div className="user-profile-info__item role">User Role:{user.role}</div>
            <div className="user-profile-info__item email">{user.email}</div>
              {(get(themeConfigs, 'isShowUserBirthday', false) && get(user, 'dateOfBirth', null))
              ? <div className="user-profile-info__item birthday">Date of Birth: {getDDMMMYYYY(user.dateOfBirth)}</div>
              : null
            }
          </div>
          { themeConfigs.isShowUserProfileSpecification ?
            <div className="specification">
              <div className="user-profile-info__title registered">Registered GP</div>
              <div className="user-profile-info__item version-back">About Showcase Stack; PulseTile version {varsionOfPulseTile}/QEWD_Ripple version {initialiseData.version}</div>
              <div className="user-profile-info__item version-front">React version {varsionOfReact}</div>
            </div>
            : null }
          <PTButton className="btn btn-theme btn-block btn-signout" aria-label="Log Out" onClick={actions.logoutStart} accessKey="S">
            <div>
              <span className="brn-text">Sign Out</span> <i className="btn-icon fa fa-sign-out" />
            </div>
          </PTButton>
        </div>
        { themeConfigs.isShowUserProfileSettings && <UserAccountPanelSettings onClose={onClose} /> }
      </div>
    )
  }
}
