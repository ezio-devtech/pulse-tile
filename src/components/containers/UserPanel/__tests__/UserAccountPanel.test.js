import React from 'react';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import configureStore from 'redux-mock-store';

import UserAccountPanel from '../UserAccountPanel';

Enzyme.configure({ adapter: new Adapter() })

const mockStore = configureStore();
const storeResource = {
  userAccount: {
    sub: '28AD8576-1948-4C84-8B5E-55FB7EE027CE',
    given_name: 'Bob',
    family_name: 'Smith',
    email: 'bob.smith@gmail.com',
    tenant: null,
    role: 'IDCR',
    roles: [
      'IDCR'
    ]
  },
  patientsInfo: {
    browserTitle: 'PulseTile',
    logoB64: 'testLogo',
    themeColor: 'green',
    title: 'PulseTile',
  },
};
const context = {
  router: {
    history: {
      test: 'test',
      goBack: () => {},
      push: () => {},
    },
  },
};
const onClick = () => console.log('test')
const storeProfilePage = mockStore(Object.assign({
  router: {
    location: {
      pathname: '/',
      search: '',
      hash: '#/profile',
    },
  } }, storeResource));
describe('Component <UserAccountPanel />', () => {
  it('should renders with props correctly', () => {
    const userAccountPanel = shallow(
      <UserAccountPanel
        store={storeProfilePage}
        onClick={onClick}
      />, { context }).dive().dive();
    expect(userAccountPanel).toMatchSnapshot();
    userAccountPanel.setContext(context);
    userAccountPanel.find('.user-profile-image').simulate('click');
    userAccountPanel.find('.name').simulate('click');
  });
});
