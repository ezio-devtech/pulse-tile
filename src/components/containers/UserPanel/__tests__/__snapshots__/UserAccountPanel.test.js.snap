// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <UserAccountPanel /> should renders with props correctly 1`] = `
<div
  className="dropdown-user dropdown-menu-right dropdown-menu"
>
  <div
    className="user-profile-image"
    onClick={[Function]}
  >
    <div
      className="img"
    >
      <img
        alt=""
        src="/images/user.jpg"
      />
    </div>
  </div>
  <div
    className="user-profile-info"
  >
    <div
      className="user-profile-info__item name"
      onClick={[Function]}
    >
      Bob
       
      Smith
    </div>
    <div
      className="user-profile-info__descr"
    >
      <div
        className="user-profile-info__item role"
      >
        User Role:
        IDCR
      </div>
      <div
        className="user-profile-info__item email"
      >
        bob.smith@gmail.com
      </div>
    </div>
    <div
      className="specification"
    >
      <div
        className="user-profile-info__title registered"
      >
        Registered GP
      </div>
      <div
        className="user-profile-info__item version-back"
      >
        About Showcase Stack; PulseTile version 
        1.0.0
        /QEWD_Ripple version 
        2.32.1
      </div>
      <div
        className="user-profile-info__item version-front"
      >
        React version 
        15.6.2
      </div>
    </div>
    <PTButton
      className="btn btn-theme btn-block btn-signout"
      onClick={[Function]}
    >
      <div>
        <span
          className="brn-text"
        >
          Sign Out
        </span>
         
        <i
          className="btn-icon fa fa-sign-out"
        />
      </div>
    </PTButton>
  </div>
  <Connect(UserAccountPanelSettings) />
</div>
`;

exports[`Component <UserAccountPanel /> should renders with props correctly with isShowUserProfileSettings  1`] = `
<div
  className="dropdown-user dropdown-menu-right dropdown-menu"
>
  <div
    className="user-profile-image"
    onClick={[Function]}
  >
    <div
      className="img"
    >
      <img
        alt=""
        src="/images/user.jpg"
      />
    </div>
  </div>
  <div
    className="user-profile-info"
  >
    <div
      className="user-profile-info__item name"
      onClick={[Function]}
    >
      Bob
       
      Smith
    </div>
    <div
      className="user-profile-info__descr"
    >
      <div
        className="user-profile-info__item role"
      >
        User Role:
        IDCR
      </div>
      <div
        className="user-profile-info__item email"
      >
        bob.smith@gmail.com
      </div>
    </div>
    <div
      className="specification"
    >
      <div
        className="user-profile-info__title registered"
      >
        Registered GP
      </div>
      <div
        className="user-profile-info__item version-back"
      >
        About Showcase Stack; PulseTile version 
        1.0.0
        /QEWD_Ripple version 
        2.32.1
      </div>
      <div
        className="user-profile-info__item version-front"
      >
        React version 
        15.6.2
      </div>
    </div>
    <PTButton
      className="btn btn-theme btn-block btn-signout"
      onClick={[Function]}
    >
      <div>
        <span
          className="brn-text"
        >
          Sign Out
        </span>
         
        <i
          className="btn-icon fa fa-sign-out"
        />
      </div>
    </PTButton>
  </div>
  <Connect(UserAccountPanelSettings) />
</div>
`;
